src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -613,7 +613,7 @@ public FieldFrame getParent() {
          * @return children of this frame.
          */
         public Set<FieldFrame> getChildren() {
-            return Collections.unmodifiableSet(children);
+            return new HashSet<>(children);
```

Comment:
I asked this question somewhere.

How ca we instruct pitest to not mutate  `Collections.unmodifiableSet` better to keep code restrictive on mutability, even nobody even tried to change this collection.